post-processing/test_post_processing.py

```diff
@@ -0,0 +1,132 @@
+    sombrero_bench_path = str((post_processing_dir / "../benchmarks/examples/sombrero/sombrero.py").resolve())
+
+    # get test log file paths
+    perflog_path = os.path.join(os.getcwd(),"perflogs/default/default/")
+    sombrero_incomplete_log = "SombreroBenchmarkIncomplete.log"
+    sombrero_incomplete_log_path = os.path.join(perflog_path, sombrero_incomplete_log)
+
+    # look for existing log files
+    sombrero_logs = []
+    sombrero_log_re = re.compile(r"^SombreroBenchmark\_\w{8}\.log$")
+    for subdirs, dirs, files in os.walk(perflog_path):
+        sombrero_logs = list(filter(sombrero_log_re.match, files))
+
+    try:
+        # to save time don't re-generate test log files if they already exist
+        if not sombrero_logs:
+            # use sombrero example to generate new test perflog file
+            run_benchmark(sombrero_bench_path)
+            benchmark_run = True
+            # look for log files again
+            for subdirs, dirs, files in os.walk(perflog_path):
+                sombrero_logs = list(filter(sombrero_log_re.match, files))
+        # check that the log files exist
+        if sombrero_logs:
+            # arbitrarily pick one of the log file names
+            sombrero_log_path = os.path.join(perflog_path, sombrero_logs[0])
+        # if the log files haven't been created, something went wrong
+        else:
+            assert False
+
+        # as above, only re-generate log file if it doesn't exist
+        if not os.path.exists(sombrero_incomplete_log_path):
+            # copy and modify regular log file
+            shutil.copyfile(sombrero_log_path, sombrero_incomplete_log_path)
+            required_field_index = 0
+            for line in fileinput.input(sombrero_incomplete_log_path, inplace=True):
+                # find index of column to be removed
+                if fileinput.isfirstline():
+                    LOG_FIELDS = line.strip().split("|")
+                    required_field_index = LOG_FIELDS.index(REQUIRED_FIELD)
+                # remove a required field
+                new_line = remove_field_from_perflog(line, required_field_index)
+                # inline replacement
+                print(new_line, end="\n")
+        # if the log file hasn't been created, something went wrong
+        if not os.path.exists(sombrero_incomplete_log_path):
+            assert False
+
+        # check incomplete log is missing required fields
+        try:
+            post.read_perflog(sombrero_incomplete_log_path)
+        except KeyError:
+            assert True
+        # if an error hasn't been thrown, something went wrong
+        else:
+            assert False
+
+        # get dataframe from complete perflog
+        df = post.read_perflog(sombrero_log_path)
+
+    finally:
+        # clean unnecessary files and folders
+        benchmark_cleanup(benchmark_run, remove_test_logs)
+
+    # check example perflog file is read appropriately
+    assert df.columns.tolist() == EXPECTED_FIELDS

```

Comment:
Also check the content of the columns. Some of them will be expected to be empty.